src/consensus/aft/raft.h

```diff
@@ -1167,6 +1174,24 @@ namespace aft
         r.idx,
         r.prev_idx);
 
+      if (is_new_follower)
+      {
+        if (state->last_idx != r.prev_idx)
+        {
+          LOG_DEBUG_FMT(
+            "New follower received first append entries with mismatch - "
+            "rolling back from {} to {}",
+            state->last_idx,
+            r.prev_idx);
+          rollback(r.prev_idx);
```

Comment:
This shouldn't ever happen but could we `throw` in `rollback()` if we rollback before `commit_idx`?

Reply:
Yeah, I've added this as `TODO` on this PR.

Reply:
I've added this in `rollback()`, though as a `LOG_FAIL` rather than a throw. I think we want to avoid crashing in this case if we 
can so opted for logging, but in practice this likely means something later breaks because we're in an unexpected state. Does anyone have a strong argument for throwing here instead?

Reply:
I'd say `throw` as we would be able to detect the error from a unit test rather than logging an error without failing the test. Also, I think most of the Raft logic (`recv_message()`) is under a big `try {} catch {}`.

Reply:
I don't think we should throw here unless we're sure this is not recoverable, which it may be. It's good to fail early in tests, but in production we want to be defensive.